### DISCUSSION TITLE
Answered/Unanswered discussions in GraphQL API

### DISCUSSION THREAD

**Discussion by profgrammer** [community/community#3925] [Discussions]
Hi, I'd like to get the total number of answered/unanswered discussions on a repository using the GraphQL API. There seems to be no efficient way of doing this, since we can fetch at most 100 discussions at a time. Thanks!

**Comment by PyvesB** [comment]
To expand a little on a possible use-case, this information would for example be useful to create new Shields.io badges. See our [relevant issue](https://github.com/badges/shields/issues/6047) for more information, but the idea would be to allow folks to display badges such as these on their READMEs:
![](https://img.shields.io/badge/discussions-19%20answered-blue)
![](https://img.shields.io/badge/discussions-3%20unanswered-blue)
![](https://img.shields.io/badge/discussions-19/22%20answered-blue)

**Comment by rickstaa** [comment]
Another use case would be to display it on the [github-readme-stats](https://github.com/anuraghazra/github-readme-stats) stats card. For more information, see https://github.com/anuraghazra/github-readme-stats/issues/1991. I, therefore, think it would be great to have the `totalCount` object under a `user.discussions` object.

**Reply by rickstaa** [reply]
@profgrammer, @PyvesB I just found the following syntax to find the number of answered and created discussions for a user:

```graphql
query { 
  viewer { 
    login
    repositoryDiscussions{
      totalCount
    }
    repositoryDiscussionComments(onlyAnswers: true) {
      totalCount
    }
  }
}
```

The repository discussion information you are looking for is, unfortunately not yet easily available. 😞

**Comment by mgriffin** [comment]
Good news @profgrammer @PyvesB @rickstaa!

We've [recently released](https://github.blog/changelog/2023-10-10-filter-answered-discussions-through-the-graphql-api/) a filter for answered discussions, which you can use to return only answered discussions, only unanswered discussions, or all discussions. This means that you can use something like this to get the count of answered questions for a specific repository:

```
{
  repository(owner: [owner], name: [repository]) {
    discussions(first: 100, answered: true) {
      totalCount
    }
  }
}
```

[The Discussions GraphQL docs](https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions#repositorydiscussions) have all the information.
